releases: add TiDB 3.0.19 release notes (#3952)
pingcap/docs: releases/release-3.0.19.md

---
title: TiDB 3.0.19 Release Notes
---

# TiDB 3.0.19 Release Notes

Release date: September 25, 2020

TiDB version: 3.0.19

## Improvements

+ TiDB

    - Mitigate the impact of failure recovery on QPS [#19764](https://github.com/pingcap/tidb/pull/19764)

## Bug Fixes

+ TiDB

    - Fix the `slow_query` error when the `slow-log` file does not exist [#20050](https://github.com/pingcap/tidb/pull/20050)
    - Add the privilege check for `SHOW STATS_META`, `SHOW STATS_BUCKET` [#19759](https://github.com/pingcap/tidb/pull/19759)
    - Forbid changing the decimal type to the integer type [#19681](https://github.com/pingcap/tidb/pull/19681)

> 

Reviewer: ```suggestion
Tools
- Lightning
    - Fix the issue that break import immediately when detect non-UTF-8 data in strict-mode  [#378](https://github.com/pingcap/tidb-lightning/pull/378)
```

## Others

+ TiDB

    - Fix the issue that alter enum/set type does not check constraint [#20045](https://github.com/pingcap/tidb/pull/20045)
    - Fix the panic that tidb-server does not release table locks [#20021](https://github.com/pingcap/tidb/pull/20021)
    - Fix the behavior of rewrite ScalarFunction IsTure . [#19901](https://github.com/pingcap/tidb/pull/19901)
    - Support adjusting the concurrency on the `union` operator [#19885](https://github.com/pingcap/tidb/pull/19885)

> + TiKV

Reviewer 2: lgtm

>     - Set `sync-log` to `true` always [#8636](https://github.com/tikv/tikv/pull/8636)

Reviewer 2: Move to `improvements`.

>     - Fix the bug that TiKV panics when parsing responses with missing reason phrases [#8540](https://github.com/tikv/tikv/pull/8540)

Reviewer 3: Move to `bug fixes`

+ PD

>     - Add an alert rule for restart [#2789](https://github.com/pingcap/pd/pull/2789)

Reviewer 3: move to `Improvements`

---
title: TiDB 3.0.19 Release Notes
---

# TiDB 3.0.19 Release Notes

Release date: September 25, 2020

TiDB version: 3.0.19

## Compatibility Changes

+ PD

    - Change the import path from `pingcap/pd` to `tikv/pd` [#2779](https://github.com/pingcap/pd/pull/2779)
    - Change the copyright information from `PingCAP, Inc` to `TiKV Project Authors` [#2777](https://github.com/pingcap/pd/pull/2777)

## Improvements

+ TiDB

>     - Mitigate the impact of failure recovery on QPS [#19764](https://github.com/pingcap/tidb/pull/19764)

Reviewer 4: ```suggestion
    - Mitigate the impact of failure recovery on QPS performance [#19764](https://github.com/pingcap/tidb/pull/19764)
```

>     - Support adjusting the concurrency on the `union` operator [#19885](https://github.com/pingcap/tidb/pull/19885)

Reviewer 4: ```suggestion
    - Support adjusting the concurrency of the `union` operator [#19885](https://github.com/pingcap/tidb/pull/19885)
```

+ TiKV

>     - Set `sync-log` to `true` always [#8636](https://github.com/tikv/tikv/pull/8636)

Reviewer 4: ```suggestion
    - Set `sync-log` to `true` by default [#8636](https://github.com/tikv/tikv/pull/8636)
```

Reviewer 5: In fact, before this option is true by default, now we have removed this configuration item, so it always appears as true

Reviewer 4: Thanks! So it is safe to say "set it to true permanently", right?

>     - Set `sync-log` to `true` as an nonadjustable value [#8636](https://github.com/tikv/tikv/pull/8636)

Reviewer 4: ```suggestion
    - Set `sync-log` to `true` as a nonadjustable value [#8636](https://github.com/tikv/tikv/pull/8636)
```

+ PD

>     - Add an alert rule for restart [#2789](https://github.com/pingcap/pd/pull/2789)

Reviewer 4: ```suggestion
    - Add an alert rule for PD restart [#2789](https://github.com/pingcap/pd/pull/2789)
```

## Bug Fixes

+ TiDB

>     - Fix the `slow_query` error when the `slow-log` file does not exist [#20050](https://github.com/pingcap/tidb/pull/20050)

Reviewer 4: ```suggestion
    - Fix the query error that occurs when the `slow-log` file does not exist [#20050](https://github.com/pingcap/tidb/pull/20050)
```

>     - Add the privilege check for `SHOW STATS_META`, `SHOW STATS_BUCKET` [#19759](https://github.com/pingcap/tidb/pull/19759)

Reviewer 4: ```suggestion
    - Add the privilege check for `SHOW STATS_META` and `SHOW STATS_BUCKET` [#19759](https://github.com/pingcap/tidb/pull/19759)
```

    - Forbid changing the decimal type to the integer type [#19681](https://github.com/pingcap/tidb/pull/19681)

>     - Fix the issue that alter enum/set type does not check constraint [#20045](https://github.com/pingcap/tidb/pull/20045)

Reviewer 4: @wjhuang2016 
```suggestion
    - Fix the issue that there is no check for constraint when altering the `ENUM`/`SET` type column [#20045](https://github.com/pingcap/tidb/pull/20045)
```

>     - Fix the panic that tidb-server does not release table locks [#20021](https://github.com/pingcap/tidb/pull/20021)

Reviewer 6: Fix a bug that tidb-server does not release table locks after a panic.

>     - Fix the panic that tidb-server does not release table locks [#20021](https://github.com/pingcap/tidb/pull/20021)

Reviewer 4: ```suggestion
    - Fix the bug that tidb-server does not release table locks after a panic [#20021](https://github.com/pingcap/tidb/pull/20021)
```

>     - Fix the behavior of rewrite ScalarFunction IsTure . [#19901](https://github.com/pingcap/tidb/pull/19901)

Reviewer 6: Fix a bug that "OR" operation is not handled correctly in where clause

>     - Fix the behavior of rewrite ScalarFunction IsTure . [#19901](https://github.com/pingcap/tidb/pull/19901)

Reviewer 4: ```suggestion
    - Fix the bug that the `OR` operation is not handled correctly in the `WHERE` clause [#19901](https://github.com/pingcap/tidb/pull/19901)
```

>     - Fix the bug that the `OR` operation is not handled correctly in the `WHERE` clause [#19901](https://github.com/pingcap/tidb/pull/19901)

Reviewer 4: ```suggestion
    - Fix the bug that the `OR` operator is not handled correctly in the `WHERE` clause [#19901](https://github.com/pingcap/tidb/pull/19901)
```

+ TiKV

    - Fix the bug that TiKV panics when parsing responses with missing reason phrases [#8540](https://github.com/tikv/tikv/pull/8540)

+ Tools

    + TiDB Lightning

>         - Fix the issue that break import immediately when detect non-UTF-8 data in strict-mode  [#378](https://github.com/pingcap/tidb-lightning/pull/378)

Reviewer 4: ```suggestion
        - Fix the issue that the TiDB Lightning process does not exit in time when encountering illegal UTF characters in the strict mode [#378](https://github.com/pingcap/tidb-lightning/pull/378)
```
